domain/learningandworkprogress/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/learningandworkprogress/induction/service/InductionScheduleService.kt

```diff
@@ -45,6 +46,32 @@ class InductionScheduleService(
       }
   }
 
+  /**
+   * Re-schedule the prisoner's Induction Schedule.
+   *
+   * Unless the Induction Schedule is already COMPLETED or PENDING_INITIAL_SCREENING_AND_ASSESSMENTS_FROM_CURIOUS the
+   * status is set to SCHEDULED and a new deadline date is set.
+   * The new deadline date is set as it were a new Induction Schedule via the [InductionScheduleDateCalculationService]
+   *
+   * Throws [InductionScheduleNotFoundException] ff the prisoner does not have an [InductionSchedule].
+   */
+  fun reschedulePrisonersInductionSchedule(prisonNumber: String, prisonerAdmissionDate: LocalDate): InductionSchedule {
+    val inductionSchedule = inductionSchedulePersistenceAdapter.getInductionSchedule(prisonNumber)
+      ?: throw InductionScheduleNotFoundException(prisonNumber)
+
+    return with(inductionSchedule) {
+      // Validate the status transition
+      inductionScheduleStatusTransitionValidator.validate(prisonNumber, scheduleStatus, SCHEDULED)
+
+      // A rescheduled Induction Schedule must have its due date set as if it were a new Induction Schedule based on the prisoner's admission date
```

Comment:
This is a key bit - the new deadline date is the date as if it were a new Induction Schedule using the date as calculated by the `InductionScheduleDateCalculationService` (and even though we are not creating a new Induction Schedule this seemed like the easiest way to do it)